core/services/relay/evm/mercury/wsrpc/cache/cache_set.go

```diff
@@ -0,0 +1,123 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+// CacheSet holds a set of mercury caches keyed by server URL
+type CacheSet interface {
+	services.Service
+	Get(ctx context.Context, client Client) (Fetcher, error)
+}
+
+var _ CacheSet = (*cacheSet)(nil)
+
+type cacheSet struct {
+	sync.RWMutex
+	services.StateMachine
+
+	lggr   logger.Logger
+	caches map[string]Cache
+
+	latestPriceTTL time.Duration
+	maxStaleAge    time.Duration
+}
+
+func NewCacheSet(cfg Config) CacheSet {
+	return newCacheSet(cfg)
+}
+
+func newCacheSet(cfg Config) *cacheSet {
+	return &cacheSet{
+		sync.RWMutex{},
+		services.StateMachine{},
+		cfg.Logger.Named("CacheSet"),
+		make(map[string]Cache),
+		cfg.LatestReportTTL,
+		cfg.MaxStaleAge,
+	}
+}
+
+func (cs *cacheSet) Start(context.Context) error {
+	return cs.StartOnce("CacheSet", func() error {
+		return nil
+	})
+}
+
+func (cs *cacheSet) Close() error {
+	return cs.StopOnce("CacheSet", func() error {
+		cs.Lock()
+		defer cs.Unlock()
+		caches := make([]io.Closer, len(cs.caches))
+		var i int
+		for _, c := range cs.caches {
+			caches[i] = c
+			i++
+		}
```

Comment:
Can you swap in `maps.Values`?
```suggestion
		caches := maps.Values(cs.caches)
```

Reply:
Cool, didn't know about this

Reply:
It's not fully std lib yet, since they are still trying to sort out iterators first, but it's been 'stable' in /exp for a while now.